toucan_connectors/mongo/mongo_connector.py

```diff
@@ -74,29 +91,63 @@ def uri(self):
             user_pass += '@'
         return ''.join(['mongodb://', user_pass, f'{self.host}:{self.port}'])
 
+    def validate_collection(self, client, collection):
+        if collection not in client[self.database].list_collection_names():
+            raise UnkwownMongoCollection(f'Collection {collection} doesn\'t exist')
+
     def get_df(self, data_source):
         client = pymongo.MongoClient(self.uri, ssl=self.ssl)
-
-        if data_source.collection not in client[self.database].list_collection_names():
-            raise UnkwownMongoCollection(f'Collection {data_source.collection} doesn\'t exist')
-
+        self.validate_collection(client, data_source.collection)
         col = client[self.database][data_source.collection]
 
-        if isinstance(data_source.query, str):
-            data_source.query = {'domain': data_source.query}
-        data_source.query = handle_missing_params(data_source.query, data_source.parameters)
-        data_source.query = nosql_apply_parameters_to_query(data_source.query,
-                                                            data_source.parameters)
-        data = []
-        if isinstance(data_source.query, dict):
-            data = col.find(data_source.query)
-        elif isinstance(data_source.query, list):
-            data = col.aggregate(data_source.query)
+        data_source.query = complete_query(data_source.query,
+                                           data_source.parameters)
+        data = col.aggregate(data_source.query)
         df = pd.DataFrame(list(data))
-
         client.close()
         return df
 
+    def get_df_and_count(self, data_source, limit):
+        if isinstance(data_source.query, dict):
+            data_source.query = [{'$match': data_source.query}]
+
+        if limit is not None:
```

Comment:
It seems odd to use 2 queries to perform both "get" and "count" since mongo cursors are supposed to keep track of resultset total count even if a `$limit` was applied. Unfortunately, it seems that the `count()` API of the `Cursor` class in the python binding is deprecated in favor of something a bit less useful in our case.

But maybe we could just use `$facet` instead ? For instance:

```python
pipeline = [
    {'$match': {'mycol': {'$gt': 0}}},
    {'$limit': 2},
]

facet = {
    "$facet": {
        "totalCount": [
            {"$count": "count"}
        ],
        "results": pipeline,
    },
}

rset = db['my-collection'].aggregate([facet])
```

will return something like:

```python
[{'results': [{'Label': 'Label 1',
               'mycol': 10,
               '_id': ObjectId('5cdae74e57c59326d22854ba')},
              {'Label': 'Label 2',
               'mycol' 13,
               '_id': ObjectId('5cdae74e57c59326d22854bb')}],
  'totalCount': [{'count': 5}]}]
```

i.e. : there's 5 documents matching the pipeline but since we use `$limit` the resultset itself only contains 2 items.